### ISSUE TITLE
ES256Algorithm Returns Result<bool>.Ok() Even If  this.PublicKey.VerifyData() Returns False 

### ISSUE THREAD

**Issue by DandiAndy** [angusmillar/SmartHealthCard#7]
Hi @angusmillar,

I noticed an issue in the ES256Algorithm. It looks to return Result<bool>.OK(...) regardless of the outcome of the verification of the payload/signature.

https://github.com/angusmillar/SmartHealthCard/blob/a600df729d3408694258ff6ded81246faaab38b4/SmartHealthCard.Token/Algorithms/ES256Algorithm.cs#L66-L72

I don't see anything in the JwsSignatureValidator that reacts to that boolean passed back either.

https://github.com/angusmillar/SmartHealthCard/blob/a600df729d3408694258ff6ded81246faaab38b4/SmartHealthCard.Token/JwsToken/JwsSignatureValidator.cs#L12-L30

Unless I'm missing something else, it looks like it's always returning that the SmartHealthCard is valid even if the payload/signature doesn't validate correctly. Let me know if I'm mistaken.

Again, thanks for your hard work. Your code has been really easy to work with.

**Comment by angusmillar** [comment]
Hi @DandiAndy , Thank you for reporting this, rather embarrassing bug. 
You were correct, the library was ignoring invalid signatures and not reporting any issue. I have just released a new Nuget package 'SmartHealthCard.Token' V1.0.2 that resolves this issue. The new version will now throw a new exception type of  `SmartHealthCardSignatureInvalidException` when the signature is found to be invalid. This new exception type inherits from the older type that was documented to be thrown, so it is backward compatible in that sense. Yet this new type now gives you a way to single out an invalid signature as opposed to any other form of error. I've also put in some bloody test cases to ensure this never happens again.

I also added another exception type 'SmartHealthCardJwksRequestException' which singles out the issue of the JWKS (public key) not being accessible, likely due to no internet connection.

I'm also going to report this bug on the Zulip https://chat.fhir.org/ 'Smart/Health-Cards' channel as it is rather critical for anyone using the library.

Thanks again and sorry for any inconvenience this has caused.
